### PR TITLE
GG-31129 Cache metrics without description

### PR DESCRIPTION
**Pull request by vmalin**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 1]
```diff
@@ -351,4 +351,60 @@
     @MXBeanDescription("Number of keys processed during index rebuilding. To get remaining number of keys for " +
         "rebuilding, subtract current value from cache size.")
     @Override public long getIndexRebuildKeysProcessed();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("Number of already rebalanced keys.")
+    @Override public long getRebalancedKeys();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("Number estimated to rebalance keys.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:360]: _Estimated total number of keys to rabalance._

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 2]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public long getEstimatedRebalancingKeys();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The number of partitions need to be cleared before actual rebalance start.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:364]: The number of partitions to be cleared before actual rebalance start_s_.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 3]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public long getRebalanceClearingPartitionsLeft();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The mean time to execute cache invokes (time in µs).")
+    @Override public float getEntryProcessorAverageInvocationTime();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The percentage of invocations on keys, which exist in cache.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:372]: The percentage of invocations on keys which[comma not needed] exist in _the_ cache.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 4]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public float getEntryProcessorHitPercentage();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The total number of invocations on keys, which exist in cache.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:376]: The total number of invocations on keys[comma not needed] which exist in _the_ cache.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 5]
```diff
@@ -351,4 +351,60 @@
...
+    /** {@inheritDoc} */
+    @MXBeanDescription("So far, the maximum time to execute cache invokes (time in µs).")
+    @Override public float getEntryProcessorMaxInvocationTime();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("So far, the minimum time to execute cache invokes (time in µs).")
+    @Override public float getEntryProcessorMinInvocationTime();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The total number of invocations on keys, which don't exist in cache.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:392]: The total number of invocations on keys[comma not needed] which don't exist in _the_ cache.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 6]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public long getEntryProcessorMisses();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The percentage of invocations on keys, which don't exist in cache.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:396]: The percentage of invocations on keys[comma not needed] which don't exist in _the_ cache.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 7]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public float getEntryProcessorMissPercentage();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The total number of cache invocations, caused update.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:400]: The total number of cache invocations _which_ caused _an_ update.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 8]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public long getEntryProcessorPuts();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The total number of cache invocations, caused no updates.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:404]: The total number of cache invocations _which_ caused no updates.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 9]
```diff
@@ -351,4 +351,60 @@
...
+    @Override public long getEntryProcessorReadOnlyInvocations();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("The total number of cache invocations, caused removal.")
```
Comment (vmalin) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:408]: The total number of cache invocations _which_ caused _a_ removal.

File: modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java [hunk 10]
```diff
@@ -351,4 +351,60 @@
     @MXBeanDescription("Number of keys processed during index rebuilding. To get remaining number of keys for " +
         "rebuilding, subtract current value from cache size.")
     @Override public long getIndexRebuildKeysProcessed();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("Number of already rebalanced keys.")
+    @Override public long getRebalancedKeys();
+
+    /** {@inheritDoc} */
+    @MXBeanDescription("Estimated total number of keys to rabalance.")
```
Comment (alievmirza) [inline modules/core/src/main/java/org/apache/ignite/mxbean/CacheMetricsMXBean.java:360]: rabalance -> rebalance

**Review by alievmirza — CHANGES_REQUESTED** [review]

**Pull request merged**